src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -1159,9 +1160,11 @@ private PendingImportObject<BrAPIObservationUnit> fetchOrCreateObsUnitPIO(Progra
             UUID trialID = trialPIO.getId();
             UUID datasetId = null;
             if (commit) {
-                datasetId = UUID.fromString(trialPIO.getBrAPIObject()
-                        .getAdditionalInfo().getAsJsonObject()
-                        .get(BrAPIAdditionalInfoFields.OBSERVATION_DATASET_ID).getAsString());
+                JsonArray datasetsJson = trialPIO.getBrAPIObject().getAdditionalInfo().getAsJsonArray(BrAPIAdditionalInfoFields.DATASETS);
```

Comment:
Potential improvement could be encapsulating this logic and the setting logic below into some sort Dataset class that operates on the BrAPITrial. Could be something for the future if you agree.

Reply:
I think this is the type of logic that would eventually be encapsulated in getters and setters once [BI-2208](https://breedinginsight.atlassian.net/browse/BI-2208) is implemented.